Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageView.swift

```diff
@@ -275,30 +276,16 @@ public struct LinkDetectionTextView: View {
     
     public var body: some View {
         Group {
-            if let displayedText {
-                Text(displayedText)
-            } else {
-                Text(message.adjustedText)
-            }
+            Text(displayText)
         }
         .foregroundColor(textColor(for: message))
         .font(fonts.body)
         .tint(tintColor)
-        .onAppear {
-            displayedText = attributedString(for: message)
-        }
-        .onChange(of: message, perform: { updated in
-            displayedText = attributedString(for: updated)
-        })
     }
     
-    private func attributedString(for message: ChatMessage) -> AttributedString {
-        var text = message.adjustedText
-        
-        // Translation
-        if let translatedText = message.textContent(for: translationLanguage) {
-            text = translatedText
-        }
+    var displayText: AttributedString {
+        let text = viewModel.textContent
```

Comment:
Having just this was enough to make it work, and I tried markdown etc and did not break anything for me.